src/base/rss/rss_session.cpp

```diff
@@ -102,6 +102,24 @@ Session::Session()
         m_refreshTimer.start(m_refreshInterval * MsecsPerMin);
         refresh();
     }
+
+    // remove legacy/corrupted settings
+    auto settingsStorage = SettingsStorage::instance();
+    settingsStorage->removeValue("Rss/streamList");
```

Comment:
Using a string list + a loop have less clutter

Reply:
>Using a string list + a loop have less clutter

Not so obvious for me... What clutter do you mean?

Reply:
something like so:
```c++
const QStringList l = {
  "Rss/streamList"
  // ...
};
for (const auto &i : l)
  settingsStorage->removeValue(i);
```

Reply:
@Chocobo1 I think it will add at least 4 extra lines compared to the current implementation. You need to list all strings anyway, so just instruct it to remove it in the same place.

Reply:
I understood what do you suggest, but I don't understand why it's better in this case.

Reply:
yeah, I noticed that, it's just a style preference and optional thing.

Reply:
One more random idea, does it make any sense to sort the list a bit (any criteria), or is it already sorted and I'm not aware?

Reply:
>or is it already sorted and I'm not aware?

Yes.
1. Legacy keys
2. Corrupted legacy keys (with new prefix `RSS`)
3. Corrupted new keys (with legacy prefix `Rss`)